diff --git a/CMakeLists.txt b/CMakeLists.txt
index f9ce7df..3bec3f1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,11 +68,13 @@ if (Qt5_POSITION_INDEPENDENT_CODE)
 endif()
 
 set(KDEPIMLIBS_LIB_VERSION "4.71.0")
+set(KMIME_LIB_VERSION "4.84.0")
+find_package(KF5CoreAddons CONFIG REQUIRED)
 find_package(KF5Akonadi ${KDEPIMLIBS_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5Contacts ${KDEPIMLIBS_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5CalendarCore ${KDEPIMLIBS_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5CalendarUtils ${KDEPIMLIBS_LIB_VERSION} CONFIG REQUIRED)
-find_package(KF5Mime ${KDEPIMLIBS_LIB_VERSION} CONFIG REQUIRED)
+find_package(KF5Mime ${KMIME_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5AkonadiNotes ${KDEPIMLIBS_LIB_VERSION} CONFIG REQUIRED)
 
 find_package(SWIG)
diff --git a/icalendar/imip.cpp b/icalendar/imip.cpp
index f4dca36..3241568 100644
--- a/icalendar/imip.cpp
+++ b/icalendar/imip.cpp
@@ -84,7 +84,7 @@ KMime::Message::Ptr createMessage( const QString &from, const QString &_to,
 
     if ( !attachment.isEmpty() ) {
       KMime::Headers::ContentDisposition *disposition =
-        new KMime::Headers::ContentDisposition( message.get() );
+        new KMime::Headers::ContentDisposition();
       disposition->setDisposition( KMime::Headers::CDinline );
       message->setHeader( disposition );
       message->contentTransferEncoding()->setEncoding( KMime::Headers::CEquPr );
@@ -103,7 +103,7 @@ KMime::Message::Ptr createMessage( const QString &from, const QString &_to,
     // Set the first multipart, the body message.
     KMime::Content *bodyMessage = new KMime::Content;
     KMime::Headers::ContentDisposition *bodyDisposition =
-      new KMime::Headers::ContentDisposition( bodyMessage );
+      new KMime::Headers::ContentDisposition();
     bodyDisposition->setDisposition( KMime::Headers::CDinline );
     bodyMessage->contentType()->setMimeType( "text/plain" );
     bodyMessage->contentType()->setCharset( "utf-8" );
@@ -115,7 +115,7 @@ KMime::Message::Ptr createMessage( const QString &from, const QString &_to,
     if ( !attachment.isEmpty() ) {
       KMime::Content *attachMessage = new KMime::Content;
       KMime::Headers::ContentDisposition *attachDisposition =
-        new KMime::Headers::ContentDisposition( attachMessage );
+        new KMime::Headers::ContentDisposition();
       attachDisposition->setDisposition( KMime::Headers::CDattachment );
       attachMessage->contentType()->setMimeType( "text/calendar" );
       attachMessage->contentType()->setCharset( "utf-8" );
diff --git a/kolabformat/mimeobject.cpp b/kolabformat/mimeobject.cpp
index d5b21e1..d2eadf1 100644
--- a/kolabformat/mimeobject.cpp
+++ b/kolabformat/mimeobject.cpp
@@ -29,6 +29,7 @@
 #include "libkolab-version.h"
 #include <QString>
 #include <QtGlobal>
+#include <KRandom>
 #include <boost/algorithm/string/predicate.hpp>
 
 
@@ -373,7 +374,7 @@ QVariant MIMEObject::Private::parseMimeMessage(const KMime::Message::Ptr &msg)
     }
     Kolab::ObjectType objectType = InvalidObject;
     if (mOverrideObjectType == InvalidObject) {
-        if (KMime::Headers::Base *xKolabHeader = msg->getHeaderByType(X_KOLAB_TYPE_HEADER)) {
+        if (KMime::Headers::Base *xKolabHeader = msg->headerByType(X_KOLAB_TYPE_HEADER)) {
             objectType = getObjectType(xKolabHeader->asUnicodeString().trimmed().toStdString());
         } else {
             Warning() << "could not find the X-Kolab-Type Header, trying autodetection" ;
@@ -390,10 +391,10 @@ QVariant MIMEObject::Private::parseMimeMessage(const KMime::Message::Ptr &msg)
     }
 
     if (!mDoOverrideVersion) {
-        KMime::Headers::Base *xKolabVersion = msg->getHeaderByType(X_KOLAB_MIME_VERSION_HEADER);
+        KMime::Headers::Base *xKolabVersion = msg->headerByType(X_KOLAB_MIME_VERSION_HEADER);
         if (!xKolabVersion) {
             //For backwards compatibility to development versions, can be removed in future versions
-            xKolabVersion = msg->getHeaderByType(X_KOLAB_MIME_VERSION_HEADER_COMPAT);
+            xKolabVersion = msg->headerByType(X_KOLAB_MIME_VERSION_HEADER_COMPAT);
         }
         if (!xKolabVersion || xKolabVersion->asUnicodeString() == KOLAB_VERSION_V2) {
             mVersion = KolabV2;
@@ -440,7 +441,7 @@ void MIMEObject::setVersion(Version version)
 
 static std::string createCid()
 {
-    return QString::fromLatin1("cid:%1@%2").arg(QString::fromLatin1(KMime::uniqueString())).arg("kolab.resource.akonadi").toStdString();
+    return QString::fromLatin1("cid:%1@%2").arg(KRandom::randomString(16)).arg("kolab.resource.akonadi").toStdString();
 }
 
 std::vector<Kolab::Attachment> convertToReferences(const std::vector<Kolab::Attachment> &attachments, std::vector<std::string> &attachmentCids)
diff --git a/mime/mimeutils.cpp b/mime/mimeutils.cpp
index 640d3e5..17a0aa3 100644
--- a/mime/mimeutils.cpp
+++ b/mime/mimeutils.cpp
@@ -153,9 +153,13 @@ KMime::Message::Ptr createMessage(const QByteArray& xKolabType, bool v3, const Q
 {
     KMime::Message::Ptr message(new KMime::Message);
     message->date()->setDateTime(KDateTime::currentUtcDateTime().dateTime());
-    message->appendHeader(new KMime::Headers::Generic(X_KOLAB_TYPE_HEADER, message.get(), xKolabType, "utf-8"));
+    KMime::Headers::Generic* h = new KMime::Headers::Generic(X_KOLAB_TYPE_HEADER);
+    h->fromUnicodeString(xKolabType, "utf-8");
+    message->appendHeader(h);
     if (v3) {
-        message->appendHeader(new KMime::Headers::Generic(X_KOLAB_MIME_VERSION_HEADER, message.get(), KOLAB_VERSION_V3, "utf-8"));
+        KMime::Headers::Generic* hv3 = new KMime::Headers::Generic(X_KOLAB_MIME_VERSION_HEADER);
+        hv3->fromUnicodeString(KOLAB_VERSION_V3, "utf-8");
+        message->appendHeader(hv3);
     }
     message->userAgent()->from7BitString(prodid);
     message->contentType()->setMimeType("multipart/mixed");
diff --git a/tests/formattest.cpp b/tests/formattest.cpp
index a3edc0e..98ab826 100644
--- a/tests/formattest.cpp
+++ b/tests/formattest.cpp
@@ -385,12 +385,12 @@ void FormatTest::testNote()
     QCOMPARE(Kolab::ErrorHandler::instance().error(), Kolab::ErrorHandler::Debug);
 
     KMime::Message::Ptr convertedNote = reader.getNote();
-    QVERIFY(convertedNote.get());
+    QVERIFY(convertedNote.data());
 
     //Parse note
     const KMime::Message::Ptr &realNote = readMimeFile( noteFileName, ok );
     QVERIFY(ok);
-    QVERIFY(realNote.get());
+    QVERIFY(realNote.data());
 
     QString expected = realNote->encodedContent();
     normalizeMimemessage(expected);
@@ -402,8 +402,8 @@ void FormatTest::testNote()
 
     //Write
     const KMime::Message::Ptr &convertedMime = Kolab::KolabObjectWriter::writeNote(realNote, version);
-    QVERIFY(convertedMime.get());
-    QVERIFY(msg.get());
+    QVERIFY(convertedMime.data());
+    QVERIFY(msg.data());
 
     QString expected2 = msg->encodedContent();
     normalizeMimemessage(expected2);
